crates/oxc_linter/src/rules/deepscan/bad_array_method_on_arguments.rs

```diff
@@ -0,0 +1,199 @@
+use oxc_ast::{
+    ast::{Expression, MemberExpression},
+    AstKind, Atom, Span,
+};
+use oxc_diagnostics::{
+    miette::{self, Diagnostic},
+    thiserror::Error,
+};
+use oxc_macros::declare_oxc_lint;
+
+use crate::{context::LintContext, rule::Rule, AstNode};
+
+#[derive(Debug, Error, Diagnostic)]
+#[error("Bad array method on arguments")]
+#[diagnostic(
+    severity(warning),
+    help(
+        "The 'arguments' object does not have '{0}()' method. If an array method was intended, consider converting the 'arguments' object to an array or using ES6 rest parameter instead."
+    )
+)]
+struct BadArrayMethodOnArgumentsDiagnostic(Atom, #[label] pub Span);
+
+/// `https://deepscan.io/docs/rules/bad-array-method-on-arguments`
+#[derive(Debug, Default, Clone)]
+pub struct BadArrayMethodOnArguments;
+
+declare_oxc_lint!(
+    /// ### What it does
+    /// This rule applies when an array method is called on the arguments object itself.
+    ///
+    /// ### Why is this bad?
+    /// The arguments object is not an array, but an array-like object. It should be converted to a real array before calling an array method.
+    /// Otherwise, a TypeError exception will be thrown because of the non-existent method.
+    ///
+    /// ### Example
+    /// ```javascript
+    /// function add(x, y) {
+    ///   return x + y;
+    /// }
+    /// function sum() {
+    ///   return arguments.reduce(add, 0);
+    /// }
+    /// ```
+    BadArrayMethodOnArguments,
+    correctness,
+);
+
+impl Rule for BadArrayMethodOnArguments {
+    fn run<'a>(&self, node: &AstNode<'a>, ctx: &LintContext<'a>) {
+        if let AstKind::MemberExpression(member_expr) = node.get().kind() 
+            && let Some(reference) = member_expr.object().get_identifier_reference()
+            && reference.name == "arguments"
```

Comment:
nitpick: try `arguments` -> parent is MemberExpression? -> parent is CallExpression?

Given member expression is the most common expression, this becomes pretty code.

Reply:
Thanks for your patient instruction.